Zend/zend_compile.c

```diff
@@ -6453,6 +6467,10 @@ void zend_compile_class_decl(zend_ast *ast) /* {{{ */
 	}
 
 	CG(active_class_entry) = ce;
+	if (extends_ast) {
+		zend_string *extends_name = zend_ast_get_str(extends_ast);
+		CG(active_class_entry_parent_name) = zend_string_copy(extends_name);
+	}
```

Comment:
Shouldn't it be NULLed out in the else case? I think it will currently not behave correctly if you have an outer class with parent and an inner (anonymous) class without.

Reply:
That sounds right. Will investigate.